### PR TITLE
Feature/confit and remove hardcoded paths

### PR DESCRIPTION
**Pull request by uglow**
- Added Confit tools (coverage, linting, mocha (removed sinon & chai as not needed at this stage), commands for CI (npm run pre-release), README.md enhancement, License file, Contributing file
- moved source code to /src to make linting easier
- webpackConfig.js now tries to use `swankyConfig.meta.src` to resolve path to source code
- src/index.js now supports the passing of the path to a swanky config file, allowing the swanky config file to be positioned anywhere on the filesystem

### CODE REVIEW

**Comment by uglow** [comment]
**Warning!**

The NPM release is at 1.9.0, but the repository release is at 1.6.2!!!

**Comment by uglow** [comment]
Pushed tag `v1.9.0` to commit e0cdb7115a61643ea452603d3af47585f2eeeb00 to get repo & NPM back in sync.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Common folders to ignore
+node_modules/*
+bower_components/*
```
Comment (Nighthawk14) [inline .eslintignore]: Is this necessary as nobody is using bower anymore? 😜

File: package.json [hunk 2]
```diff
@@ -1,49 +1,47 @@
 {
   "name": "swanky-server",
-  "description": "Swanky Docs",
-  "main": "index.js",
+  "description": "Swanky Docs Server",
+  "license": "MIT",
+  "author": "Rod Leviton",
+  "main": "src/index.js",
+  "version": "0.0.0-semantically-released",
```
Comment (Nighthawk14) [inline package.json]: Is the version right here? `0.0.0-semantically-released`

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {string | undefined} swankyConfigFilePath   Path to the swanky config file
+ * @returns {Object}                                  Configuration object containing "built" swanky config plus additional meta data, to be used by Webpack
+ */
+function getConfig(swankyConfigFilePath) {
+  let _swankyConfig = swanky.loadConfig(swankyConfigFilePath || DEFAULTS.SWANKY_CONFIG);
+
+  _swankyConfig.meta = swanky.getSiteMetadata(_swankyConfig);
+  _swankyConfig.meta.cssScopedName = DEFAULTS.CSS_SCOPED_NAME;
+
+  //console.log(_swankyConfig);
```
Comment (Nighthawk14) [inline src/index.js]: Debug?

File: test/unit/fixtures/content/foo.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Some text
```
Comment (Nighthawk14) [inline test/unit/fixtures/content/foo.md]: Should this file exists?

File: package.json (hunk 2 above)
Comment (uglow) [inline package.json]: Yep - semantic release will update package.json when a release is generated.

File: src/index.js (hunk 3 above)
Comment (uglow) [inline src/index.js]: Yeah, I think I'll need it again soon.

File: test/unit/fixtures/content/foo.md (hunk 4 above)
Comment (uglow) [inline test/unit/fixtures/content/foo.md]: Yes, this file *must* exist and must have content for the test spec to work.

**Comment by Nighthawk14** [comment]
👍

File: src/config/webpack/webpack.config.js [hunk 5]
```diff
@@ -19,16 +19,16 @@ module.exports = (config, swankyConfig) => {
...
       reset: [
         path.join(__dirname, '../reset/index.js')
       ]
     },
     output: {
-      path: path.join(basepath, 'build'),
+      path: path.join(basepath, 'build'), // TODO: Make this a variable
       filename: '[name].[hash:8].bundle.js',
-      publicPath: swankyConfig.meta.production ? swankyConfig.meta.subdirectory : '/'
+      publicPath: swankyConfig.meta.production ? swankyConfig.meta.subdirectory : '/'     // TODO: Explain subdirectory
```
Comment (rodleviton) [inline src/config/webpack/webpack.config.js]: Its clear I **do** not have a knack for naming variables

**Pull request merged**